src/js/menu/menu.js

```diff
@@ -122,6 +169,43 @@ class Menu extends Component {
     }
   }
 
+  /**
+   * Called when a `MenuItem` gets clicked or tapped.
+   *
+   * @param {EventTarget~Event} event
+   *        The `click` or `tap` event that caused this function to be called.
+   *
+   * @listens click,tap
+   */
+  handleTapClick(event) {
+    // Unpress the associated MenuButton, and move focus back to it
+    if (this.menuButton_) {
+      this.menuButton_.unpressButton();
+
+      const childComponents = this.children();
+
+      if (!Array.isArray(childComponents)) {
+        return;
+      }
+
+      const foundComponent = childComponents.find(component => {
```

Comment:
This is because we no longer have direct access to the component, we need to find the component object that matches the event?

Reply:
Yes, we lost the component reference after moving "handleTapClick" out of closure because the function will align with "handleBlur" function, and using the same way to free both of these listeners after removing of the menu item from the menu instance.

Reply:
Also I found the commit that add the check whether the clicked menu item is "CaptionSettingsMenuItem" or not to decide to call menu button's focus method

cb890a965cb0dc0df11f55eabea091b2cbb49d71

Reply:
Thanks, just wanted to confirm I was reading it right :)